readLog/src/main/java/org/corfudb/readLog/readLog.java

```diff
@@ -0,0 +1,354 @@
+        } else {
+            System.out.print(USAGE);
+            System.out.println("Return FALSE");
+            return false;
+        }
+        if (opts.get("--erase") != null) {
+            int addr = Integer.parseInt(((String) opts.get("--erase")).trim());
+            op = new Operation(Operation.OperationType.ERASE_ONE, addr);
+            use_output_file = true;
+        } else if (opts.get("--erase_tail") != null) {
+            int addr = Integer.parseInt(((String) opts.get("--erase_tail")).trim());
+            op = new Operation(Operation.OperationType.ERASE_TAIL, addr);
+            use_output_file = true;
+        } else if (opts.get("--display") != null) {
+            int flag = Integer.parseInt(((String) opts.get("--display")).trim());
+            if (flag == 0) {
+                op = new Operation(Operation.OperationType.DISPLAY, 0);
+            } else {
+                op = new Operation(Operation.OperationType.DISPLAY_ALL, 0);
+            }
+        }
+        LogHeader lh = LogHeader.newBuilder()
+                .setVersion(10)
+                .setVerifyChecksum(true)
+                .build();
+        LogHeaderSize = lh.getSerializedSize();
+        Metadata md = Metadata.newBuilder()
+                .setChecksum(0x4d5ca4cc)
+                .setLength(lh.getSerializedSize())
+                .build();
+        MetadataSize = md.getSerializedSize();
+
+        File f_in = new File(log_file_name);
+        File f_out = use_output_file ? new File(log_file_name + ".modified") : null;
+        if (f_in.canRead()) {
+            try {
+                fileStreamIn = new FileInputStream(f_in);
+                fileChannelIn = fileStreamIn.getChannel();
+                if (use_output_file) {
+                    fileStreamOut = new FileOutputStream(f_out);
+                    fileChannelOut = fileStreamOut.getChannel();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return true;
+        }
+        return false;
+    }
+
+    int readAll() {
+        int record_cnt = 0;
+        try {
+            record_cnt = processLogFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return record_cnt;
+    }
+
+    void cleanUp() {
+        try {
+            if (fileStreamIn != null) {
+                fileStreamIn.close();
+                fileStreamIn = null;
+            }
+            if (fileStreamOut != null) {
+                fileStreamOut.close();
+                fileStreamOut = null;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    static int getChecksum(byte[] buff) {
+        HashFunction hf = Hashing.crc32c();
+        HashCode hc = hf.newHasher()
+                .putBytes(buff)
+                .hash();
+        return hc.asInt();
+    }
+
+    static LogEntry buildLogEntry(LogEntry entry) {
```

Comment:
This method takes an entry and converts it to a hole, please make sure the signature is descriptive.